lib/client/doc.js

```diff
@@ -741,6 +737,19 @@ Doc.prototype._submit = function(op, source, callback) {
   });
 };
 
+// Try to normalize the op. This removes trailing skip:0's and things like that.
+Doc.prototype._normalizeOp = function(op) {
+  if (!this.type) {
```

Comment:
This isn't really "normalising", but it's common code. Open to naming improvements on the method.